Tasks/AzureRmWebAppDeployment/kuduutility.ts

```diff
@@ -0,0 +1,466 @@
+            attempts += 1;
+            try {
+                var fileContent = (await getFileContent(publishingProfile, physicalPath, fileName))['content'];
+                if(fileAction  === 'CheckFileExists') {
+                    tl.debug('Found file:  ' + fileName);
+                    defer.resolve(fileContent);
+                }
+                else if(fileAction === 'CheckFileNotExists') {
+                    tl.debug('File: ' + fileName + 'found. retry after 10 seconds. Attempt: ' + attempts);
+                    setTimeout(poll, 10000);
+                }
+                else {
+                    defer.reject(tl.loc('InvalidPollOption', fileAction));
+                }
+            }
+            catch(error) {
+                if(error.statusCode === 404) {
+                    if(fileAction  === 'CheckFileExists') {
+                        tl.debug('File ' + fileName + ' not found. rerty after 10 seconds. Attempt : ' + attempts);
+                        setTimeout(poll, 10000);
+                    }
+                    else if(fileAction === 'CheckFileNotExists') {
+                        tl.debug('File: ' + fileName + 'not found.');
+                        defer.resolve();
+                    }
+                    else {
+                        defer.reject(tl.loc('InvalidPollOption', fileAction));
+                    }
+                }
+                else {
+                    if(error.error) {
+                        defer.reject(tl.loc('FailedToGetKuduFileContentError', fileName, error.error));
+                    }
+                    else {
+                        defer.reject(tl.loc('FailedToGetKuduFileContent', fileName, error.statusCode, error.statusMessage));
+                    }
+                }
+            }
+        }
+        else {
+            tl.warning(tl.loc('PollingForFileTimeOut'));
+            defer.reject(tl.loc('ScriptStatusTimeout'));
+        }
+    }
+    poll();
+    return defer.promise;
+}
+
+async function getPostDeploymentScriptLogs(publishingProfile, physicalPath, uniqueID) {
+
+    try {
+        var stdoutLog = (await getFileContentUtility(publishingProfile, physicalPath, 'stdout_' + uniqueID + '.txt'));
+        var stderrLog = (await getFileContentUtility(publishingProfile, physicalPath, 'stderr_' + uniqueID + '.txt'));
+        var scriptReturnCode = ((await getFileContentUtility(publishingProfile, physicalPath, 'script_result_' + uniqueID + '.txt')).trim());
+    }
+    catch(error) {
+        throw Error(error);
+    }
+   if(stdoutLog) {
+        console.log(tl.loc('stdoutFromScript'));
+        console.log(stdoutLog);
+    }
+    if(stderrLog) {
+        console.log(tl.loc('stderrFromScript'));
+        if(scriptReturnCode != '0') {
+            tl.error(stderrLog);
+            throw Error(tl.loc('ScriptExecutionOnKuduFailed', scriptReturnCode, stderrLog));
+        }
+        else {
+            console.log(stderrLog);
+        }
+    }
```

Comment:
Extensions like 'composer' provides debug logs in error stream.so, we will check for return code of the script and fail for non-zero.